test_runner/regress/test_replication_start_subxid_overflow.py

```diff
@@ -0,0 +1,239 @@
+    p_cur.execute("select create_subxacts(100000)")
+
+    # Create a replica at this LSN
+    wait_for_last_flush_lsn(env, primary, env.initial_tenant, env.initial_timeline)
+    with env.endpoints.new_replica_start(origin=primary, endpoint_id="secondary") as secondary:
+        s_con = secondary.connect()
+        s_cur = s_con.cursor()
+
+        # The transaction in primary has not committed yet.
+        wait_replica_caughtup(primary, secondary)
+        s_cur.execute("select count(*) from t")
+        assert s_cur.fetchone() == (0,)
+
+        # Add more sub-xids to the same transaction in primary
+        p_cur.execute("select create_subxacts(10000)")
+
+        # The transaction still hasn't committed
+        wait_replica_caughtup(primary, secondary)
+        s_cur.execute("BEGIN ISOLATION LEVEL REPEATABLE READ")
+        s_cur.execute("select count(*) from t")
+        assert s_cur.fetchone() == (0,)
+
+        # Commit the transaction in primary
+        p_cur.execute("commit")
+
+        # Should still be invisible to the old snapshot
+        wait_replica_caughtup(primary, secondary)
+        s_cur.execute("select count(*) from t")
+        assert s_cur.fetchone() == (0,)
+
+        # Commit the REPEATABLE READ transaction in standby
+        s_cur.execute("commit")
+
+        # The transaction should now visible to new snapshot in standby
+        s_cur.execute("select count(*) from t")
+        assert s_cur.fetchone() == (200000,)
+
+# Test starting a standby from situation that there is one transaction running
+# in primary, with lots of subtransactions. Then, after the standby is running,
+# start a large number of new transcations, with lots of subtransactions each, in
+# the primary.
+#
+# XXX: This currently crashes the startup process in secondary with:
+#
+# FATAL:  too many KnownAssignedXids
+# CONTEXT:  WAL redo at 0/1895CB0 for neon/INSERT: off: 25, flags: 0x08; blkref #0: rel 1663/5/16385, blk 64
+def test_replication_start_subxid_overflow3(neon_simple_env: NeonEnv):
+    env = neon_simple_env
+
+    primary = env.endpoints.create_start(branch_name="main", endpoint_id="primary")
+
+    with primary.connect() as p_con:
+        p_cur = p_con.cursor()
+
+        p_cur.execute("create table t(pk serial primary key, payload integer)")
+        p_cur.execute(
+            """create or replace function create_subxacts(n integer) returns void as $$
+            declare
+                i integer;
+            begin
+                for i in 1..n loop
+                    begin
+                        insert into t (payload) values (0);
+                    exception
+                        when others then
+                            raise exception 'caught something';
+                    end;
+                end loop;
+           end; $$ language plpgsql"""
+        )
+
+        p_cur.execute(f"show max_connections")
+        max_connections = int(p_cur.fetchone()[0])
+
+    n_connections = max_connections - 2
```

Comment:
`max_connections - 2` is expected to be too large: test is failed with "too many connection error".
With `max_connections - 10` it works (have not tested smaller values).